### PR TITLE
Add `batch-mark-files-as-viewed` feature

### LINKED ISSUE #2444: Mark all files as viewed/not-viewed
While reviewing PRs with 10+ files, the workflow gets easier by marking files as viewed. However, when I need to re-review the PR, it gets very tedious marking all files as not-viewed before going ahead with the review.

Would be great if we had a button that would mark all files as view or not-viewed.

### PR DESCRIPTION
**Pull request by fregante**
Fixes #2444 

Test on a PR with more than 2 files, like https://github.com/sindresorhus/refined-github/pull/3006/files

### CODE REVIEW

File: source/features/batch-mark-files-as-viewed.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	delegate(document, '.js-toggle-user-reviewed-file-form', 'submit', remember);
+}
+
+function deinit(): void {
+	previousFile = undefined;
+}
+
+features.add({
+	id: __featureName__,
+	description: 'Mark/unmark multiple files as Viewed in the PR Files tab. Click on the first checkbox you want to mark/unmark and then `shift`-click another one; all the files between the two checkboxes will be marked/unmarked as Viewed.',
```
Comment (sindresorhus) [inline source/features/batch-mark-files-as-viewed.tsx:51]:
```suggestion
	description: 'Mark/unmark multiple files as “Viewed” in the PR Files tab. Click on the first checkbox you want to mark/unmark and then `shift`-click another one; all the files between the two checkboxes will be marked/unmarked as “Viewed”.',
```

**Comment by fregante** [comment]
I want to make 2 more changes:

- [ ] Add alt-click to toggle all
- [ ] Add a "in progress" indicator next to the counter, where I circle in the gif

**Comment by fregante** [comment]
I'll merge this as is and I'll leave those 2 changes in a separate issue, if anyone wants to tackle them: https://github.com/sindresorhus/refined-github/issues/3043

**Comment by b-hayes** [comment]
Really need a way to mark all deleted files as viewed in one go fi that can be added.
Or even just a way to mark ALL as viewed and claps all and just scroll through the names of such enormous PR's and untick specific files of interest.
image

**Comment by fregante** [comment]
What's the point of marking files as viewed if you haven't viewed them?

Currently RGH supports marking ALL as viewed via alt-click on a checkbox, as long as they've been loaded (the list is lazy-loaded)

**Comment by mfb2** [comment]
Is this actually implemented?  I just tried it and it doesn't work.

**Comment by arthurio** [comment]
@cheap-glitch I'm using a mac, maybe I'm trying the wrong key press combination? I tried:
- option + shift + click
- cmd + shift + click
- cmd + click
- option + click

None of those worked ¯\_(ツ)_/¯

**Comment by cheap-glitch** [comment]
> I'm using a mac, maybe I'm trying the wrong key press combination?

It's a simple `Shift` + click, like in file browsers. Just tested right now and it's working for me on Firefox (Arch Linux).

Maybe it's a browser bug, you're on Safari I presume?

**Comment by arthurio** [comment]
I'm using Chrome, but also tried Firefox... `Shift + click` doesn't work either 😢

https://www.loom.com/share/150d7721e07f4effa535706183ce34d9

**Comment by fregante** [comment]
Thanks for the video. I'm not sure either. The correct sequence is:

1. Click on first item
2. Hold shift
3. Click on other item
4. Release shift

And steps 2-4 can be repeated to continue the selection. 

Are you using the left shift? Maybe that's it. Or maybe CAPS LOCK is enabled? 😅

**Comment by arthurio** [comment]
@fregante It works great! I had some weird key binding conflicts 👍🏻 Thanks a lot for your help!

**Comment by JonHendershot** [comment]
> @fregante It works great! I had some weird key binding conflicts 👍🏻 Thanks a lot for your help!

@arthurio mind sharing what conflicts you had and how you resolved them? think i may be having the same issue

**Comment by chrisshaddad** [comment]
@arthurio - I'm struggling to get this to work on my mac, mind sharing what key bindings you changed?

**Comment by arthurio** [comment]
I'm using vimium with chrome, I just disabled it for github and it worked.

What browser do you use? What's your version of the github-refined extension?

**Comment by chrisshaddad** [comment]
@arthurio - Thank you for the quick reply! Much appreciated, I figured it out, I had to install `refined-github` first 😭🔫

**Comment by arthurio** [comment]
@chrisshaddad Haha yeah... It's not 100% obvious TBH so don't feel bad 😄 I have been using it for couple months now and I love it! Enjoy 😉

**Comment by devibrahimkarahan** [comment]
> Thanks for the video. I'm not sure either. The correct sequence is:
> 
> 1. Click on first item
> 2. Hold shift
> 3. Click on other item
> 4. Release shift
> 
> And steps 2-4 can be repeated to continue the selection.
> 
> Are you using the left shift? Maybe that's it. Or maybe CAPS LOCK is enabled? 😅

It doesnt work with Shift, Chrome 114.0.5735.198

**Comment by yakov116** [comment]
@devibrahimkarahan Don't comment on a 3 year old post. Open a new issue.

**Pull request merged**